Update plugins/fields/media/media.php to prevent deprecated message caused if $value empty.

particthistle opened the pull request:
### Summary of Changes
PHP error was encountered when test instructions below are met:

`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

If $value is null at line 86 in plugins/fields/media/media.php, then the deprecated error occurs.

Added if statement to checkValue function to check $value is set before trying to decode the string.


### Testing Instructions

- Create a custom field for articles that is a Media type, that's also set to optional (ie Required = No)
- Create an article, leaving the optional custom fields blank.
- View the article in the front end.
- Turn on Error Reporting for the site to Development so you can see the messages on screen.

(ShackOpenGraph also installs Social Images fields that are Media type and trigger the error)

### Actual result BEFORE applying this Pull Request

If there's no image in the custom field, then the code in the plugin does a check on the $value provided to the checkValue function. As there's no value, json_decode triggers 
`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

### Expected result AFTER applying this Pull Request
The checkValue function will now firstly check whether $value is null, and then only proceed with the rest of the function when $value has a value provided to it.

This in turn stops the deprecated message being generated.

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

particthistle committed:
Update media.php

json_decode(): Passing null to parameter #1 ($json) of type string is deprecated

If $value is null at line 86, then a deprecated error occurs.

Added if statement to checkValue function to check $value is set before trying to decode the string.

particthistle committed:
Merge branch '4.2-dev' into patch-6

laoneo commented on plugins/fields/media/media.php:
```diff
@@ -83,13 +83,14 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
     private function checkValue($value)
     {
-        json_decode($value);
+        if ($value) {
```
If you invert the check, then the whole functuon becomes easier to read. Like if (! $value) {return ;}

particthistle replied:
Thanks @laoneo - makes sense... please check change I made makes sense.

particthistle committed:
Incorporating suggested change.

particthistle committed:
Update media.php

Replacing whitespace that was taken out.

particthistle committed:
Update media.php

Cleaning up whitespace.

Quy commented:
I have tested this item :white_check_mark: successfully on 9e169dff65f82ce4406bf8c85a5167510c991af7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39542">issues.joomla.org/tracker/joomla-cms/39542</a>.</sub>

joomdonation committed:
[4.2] Fix warning when null is passed to  FieldsHelper::extract method (#39864)

* Fix warning when null is passed to extract method of FieldsHelper class

roland-d committed:
Reset to dev

brianteeman committed:
[4.2] Contact form banned emails and content (#39907)

* [4.2] Contact form banned emails

Adds an ltrim so that if the entries are in the form example.com; example2.com then example2.com will still be blocked

Improved the description  It is not a list of email addresses but of email addresses or parts of it (e.g. domains).

* @fedik suggestions

* more

---------

Fedik committed:
Fix tooltip for subform table layout, #39848 (#39894)

brianteeman committed:
[4.2] users modal formatting (#39891)

Spotted this minor codestyle issue

code review only

ManuelHu committed:
[4.2] Fix MailTemplate array regression from 4.2.7 (#39850)

JamesNGarrett committed:
Allow subform field value to be an array (#39842)

brianteeman committed:
Admin Module Cpanel Positions (#38633)

Fedik committed:
Fix for the title of a loaded module is not visible (#39839)

* Fix default style for "load module" plugin

* Fix default style for "load module" plugin

* Fix default style for "load module" plugin, performance

* Fix default style for "load module" plugin, deprecation

* Fix default style for "load module" plugin, remove unused

* phpcs

* Update plugins/content/loadmodule/loadmodule.xml

Co-authored-by: Brian Teeman <brian@teeman.net>

---------

Co-authored-by: Brian Teeman <brian@teeman.net>

GaryA committed:
Issue #39820 fixed warnings in com_contact (#39826)

* Issue #39820 fixed warnings in com_contact

* Update components/com_contact/tmpl/category/default_items.php

Co-authored-by: Quy <quy@nomonkeybiz.com>

* Now handle cases where user has `core.edit.own` permission but not `core.edit`
and where the current list does not have any contacts that belong to the user

* Update components/com_contact/tmpl/category/default_items.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

---------

Co-authored-by: Quy <quy@nomonkeybiz.com>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.2] Admin module dropdown (#39906)

chmst committed:
Fix display of data-function (#38959)

brianteeman committed:
[4.2] contact fields remove layout override (#37716)

joomdonation committed:
[4.2] Implement com_config dispatcher, fix ACL bug (#39879)

JackKellyUK committed:
Firefox subform drag fix (#39576)

* Firefox subform drag fix

* Reorder styling

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Moved padding above margin top

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.3] Smart Search options (#39705)

Fedik committed:
Mail template fix "reply to" (#39893)

* Mail template fix "reply to", #37473

* Update

---------

sergeytolkachyov committed:
get Database from DI in Library helper (#39276)

* get Database from DI in Library helper

* Update LibraryHelper.php

change 'DatabaseDriver' to DatabaseInterface::class

* Update libraries/src/Helper/LibraryHelper.php

Co-authored-by: Quy <quy@nomonkeybiz.com>

particthistle committed:
Merge branch '4.2-dev' into patch-6

particthistle commented:
With 4.2-dev locked for 4.3 release, rebased this PR to 4.3-dev.

joomdonation commented:
@particthistle Actually, 4.2-dev still accepts bugs fix. So you should not have to rebase to 4.3-dev.

joomdonation committed:
[4.2] Fix deprecated warnings for HTMLHelper Calendar (#39869)

brianteeman committed:
[4.2] module position string (#39924)

charvimehradu committed:
Popovers replaced by Tooltips in Category list in Cassiopeia #35467 (#39643)

Co-authored-by: Quy <quy@nomonkeybiz.com>

brianteeman committed:
[4.2] select category string (#39930)

corrects the capitalisation

brianteeman committed:
[4.2] lang string (#39923)

carlitorweb committed:
Login feedback if user do not have offline access (#39561)

brianteeman committed:
Revert 39329 due to issue in media folders of template manager (#39943)

Co-authored-by: Quy <quy@nomonkeybiz.com>

Commit:
Correct return types (#39395)

carlitorweb committed:
Discover emptystate layout messages (#39563)

joomdonation committed:
[4.2] Fix missing context for com_fields after publish/unpublish items (#39926)

* Fix missing context for com_fields after publish/unpublish items

* Use parent::getRedirectToListAppend()

* Remove return type definition

---------

joomdonation committed:
[4.2] Fix wrong data for calendar custom fields when user registration fail (#39863)

brianteeman committed:
[4.2] Normalise extension names (#39954)

Co-authored-by: Fedir Zinchuk <getthesite@gmail.com>

korenevskiy committed:
Fix TinyMCE images directory options if '/administrator/images' exists (#39962)

LukasHH committed:
Email Cloaking fires when @ in filename (image) #39786 (#39932)

* Email Cloaking fires when @ in filename (image)
Fixes #39786

* Revert "Email Cloaking fires when @ in filename (image)"

This reverts commit 24d43bbccf381010c47f70e5fcf561924899f28a.

* Email Cloaking fires when @ in filename (image)
Fixes #39786

* Email Cloaking fires when @ in filename (image)
Fixes #39786

* Email Cloaking fires when @ in filename (image)
Fixes #39786

---------

Hackwar committed:
Fixing URL to contact VCF (#39938)

dgrammatiko committed:
Fix extra space (#39989)

Fedik committed:
Improve Finder ajax error handling (#39973)

* Improve error handling for Finder indexer.js

* Improve error handling for Finder autocomplete

* jscs

---------

korenevskiy committed:
[plgTinyMCE] Error correction. Now you can drag and drop images into the material so that the image is saved to a custom directory. (#39957)

* Update plugin.es5.js

* Update plugin.es5.js

* Update plugin.es5.js

* Update plugin.es5.js

* Update plugin.es5.js

* Update plugin.es5.js

---------

alikon committed:
[4][com_actionlog] - log event reset password (#39435)

* Update Joomla.php

* event

* Update Joomla.php

* cs

* onUserAfterResetRequest

* onUserAfterResetRequest

* onUserBeforeResetRequest, onUserBeforeResetComplete

---------

richard67 committed:
Don't append but set the "Content-Encoding" header value in htaccess.txt (#39205)

joomla-translation-bot committed:
Translation Update (#39876)

richard67 committed:
[4.2] Update deleted files and folders lists in script.php for upcoming 4.2.9 (#39998)

fancyFranci committed:
Fix since tags

fancyFranci committed:
Joomla! 4.2.9 Release Candidate 1

fancyFranci committed:
Reset to dev

Fedik committed:
PHP polyfills as explicit dependency (#39968)

fancyFranci committed:
Joomla! 4.2.9 Stable

fancyFranci committed:
Reset to dev

particthistle committed:
Merge branch 'joomla:4.2-dev' into patch-6

obuisard commented:
It looks like something went wrong with the files associated with the PR. Can you fix it on your end Patrick @particthistle? Thanks, I appreciate it.

particthistle commented:
Was nudged on this during PBF. Aside from botching the rebasing of the PR, and being away 3 months due to finding a new house and moving, the problem preventing the code being merged is more due to the entire Custom Fields code being refactored for Joomla 4.3 from what I have worked out. Had it been actioned back in February it would not have had an issue.

https://github.com/joomla/joomla-cms/blob/4.4-dev/plugins/fields/media/src/Extension/Media.php shows where the code has moved to.

In trying to replicate the problem with 4.3.3 and 4.3.4.rc1 it appears to have been resolved in the refactoring process.

If it does turn out that it needs to be fixed (scenario may be where data has been imported from J3 to J4, and the initial values for the data is NULL instead of a correct default blank value) then the code to add into a new PR before [Line 89](https://github.com/joomla/joomla-cms/blob/d20c4557cea681d02aeceee97e9ece795961da2f/plugins/fields/media/src/Extension/Media.php#L89) is:

`        // If the value is null return as json_decode will cause a deprecated issue with json_decode
        if (!$value)
        {
            return;
        }`

However as I can't replicate the problem with the above testing instructions, I'm closing this PR at the moment.

If anyone wants to walk me through the issue, or help with creating some "Github PR for Joomlers" content, reach out on Mattermost.
